### ISSUE TITLE
Runtime project dependencies are not included in the shadowJar

### ISSUE THREAD

**Issue by jpdamon** [johnrengelman/shadow#152]
I have a multi-module build with both compile and runtime dependencies on other projects:

``` gradle
dependencies {
    compile project(":common")

    runtime project(":agents:mqtt")
}
```

I would expect both to end up in the shadow jar when running `gradle shadowJar`, but only the "compile project" ones do.

As a workaround, `gradle build shadowJar` will cause both compile and runtime project dependencies to be bundled.

**Comment by johnrengelman** [comment]
`shadowJar` by default depends on the `runtime` scope, so all it's dependencies should be included. if you have a reproducible example you can post, I'd be happy to look.

**Comment by ddimtirov** [comment]
I have posted a reproducible case in this thread: https://discuss.gradle.org/t/gradle-doesnt-rebuild-submodules-on-resolving-configurations/13468/3

**Comment by johnrengelman** [comment]
Thanks @ddimtirov.
I was able to reproduce the error and I see what the problem is.
There issue is coming from this line here: https://github.com/johnrengelman/shadow/blob/master/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar.java#L68-L71

That line resolves the configuration and sets the resulting file paths as an input for the task but in this process it's losing the context of the task dependency that produced that file.

I believe this fix is that we need to declare this method: https://github.com/johnrengelman/shadow/blob/master/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/tasks/ShadowJar.java#L279-L281 as dependencies for the task.

In fact adding this line to the `build.gradle` fixed the issue:

```
shadowJar.dependsOn shadowJar.configurations
```

**Comment by johnrengelman** [comment]
Fixed with 5013695
